adapters/repos/db/lsmkv/segment_group_compaction.go

```diff
@@ -325,7 +326,15 @@ func (sg *SegmentGroup) replaceCompactedSegments(old1, old2 int,
 		return fmt.Errorf("precompute segment meta: %w", err)
 	}
 
+	// wait for flushing to complete before acquiring the maintenance lock, this
+	// can help avoid blocking user requests
+	sg.waitForFlushingToComplete(10*time.Millisecond, 60*time.Second)
+
+	beforeMaintenanceLock := time.Now()
 	sg.maintenanceLock.Lock()
+	if time.Since(beforeMaintenanceLock) > 50*time.Millisecond {
+		sg.logger.WithField("duration", time.Since(beforeMaintenanceLock)).Warn("compaction took longer than 50ms to acquire maintenance lock")
```

Comment:
just wondering if the `warn` level is not to strict here...

Reply:
You are right, the others use 100ms and `debug`.

I originally had `50ms` and `WARN` everywhere during testing, looks like I missed this one.

Reply:
Fixed in latest commit.